### ISSUE TITLE
Scan only spawn points not sequental map cells

### ISSUE THREAD

**Issue by azotov** [AHAAAAAAA/PokemonGo-Map#1056]
Pokemons always appear in fixed points, for each wild pokemon there is spawnpoint_id key in Niantic server response. In the same response you can get the list of all nearby spawn points with spawnpoint_id's and lat/lon. Every spawnpoint spawns new pokemon at fixed time every 60 minutes, these pokemons disappear after 15 minutes. For example spawn point with id=46b54a8d9d9 spawns new pokemon exactly at 37:56.292 every hour.

More efficient scan algorithm will be the following:

1. at runserver.py start create new sqlite table: CREATE TABLE "spawnpoint" ("spawnpoint_id" VARCHAR(255) NOT NULL PRIMARY KEY, "latitude" REAL NOT NULL, "longitude" REAL NOT NULL, "spawn_time" DATETIME)
2. at runserver.py start get all spawnpoints near our point of interest with existing cell walk algorithm and save them to spawnpoint table with spawn_time equal NULL
3. use first thread to check cells with spawnpoints which are near our point of interest and have spawn_time == NULL every 10 minutes. if pokemon spawn is detected at the point, then update spawn_time with "disappear_time - 15 minutes".
4. use second thread to check cells with spawnpoints which are near our point of interest with known spawn_time at 30 seconds after spawn_time every 60 minutes to find which pokemon have been spawned this time.

This algorithm enhancement will help to reduce number of request per hour and greatly increase area monitored with one account.

**Comment by JZ6** [comment]
Spawn points only apply to specific Pokemon right

**Comment by azotov** [comment]
For example spawn point with id==46b54a8d9d9 spawned two Nidoran♀, two Drowzee, one Shellder and one Jynx in last six hours. The probabilities of spawning specific pokemons for each point maybe depend on theirs surrounding (rivers/parks/etc) and should be fixed values.

**Comment by JZ6** [comment]
I see that is interesting , I'm gonna check logs to see if there's a pattern anywhere.

**Comment by Biztactix-Ryan** [comment]
Definitely Spawn Points spawn different pokemon, And on the hour as described by Op... I think this would be great !
Very efficient

**Comment by roland3** [comment]
I also thought about it and found this issue.
It would also allow us to show spawn points on map so we could know when and where to expect next spawns. I guess timers would make the map quite messy, but something like colored points (say, red to green) could be a neat solution.

**Comment by commanderjsoc** [comment]
I would also like to see where each of the spawnpoints are on the map

**Comment by WeeschDoONi** [comment]
I'm working on something in that direction. I used the information saved in the .db files to gather spawn points and the pokemons which spawned there. Current screenshot:

![spawns](https://cloud.githubusercontent.com/assets/20650430/17118356/7bed47ae-52c0-11e6-83c4-8e8dafdd19c1.png)

**Comment by commanderjsoc** [comment]
I would love to see what you create! make sure to submit a pull request!

**Comment by ximex** [comment]
@WeeschDoONi looks great. could you change the list in the bubble to a piechart?

**Comment by commanderjsoc** [comment]
Great Idea @ximex

**Comment by commanderjsoc** [comment]
@WeeschDoONi Do you have your code on github? I would like to try it out

**Comment by WeeschDoONi** [comment]
@ximex I'm not that into JSON/Javascript. But if there is a library which turns JSON-Objects into piecharts it shouldn't be a problem.

@commanderjsoc I just threw it together over the last couple of hours, I don't think it's following the style of the project so it's just on my computer at the moment. I might fork the project if I find the time to keep working on it.

**Comment by azotov** [comment]
I use now a small php script which generates kml file with spawn point but having spawns in browser is much more convenient, so I am waiting @WeeschDoONi pull request too.
Some possible improvements:
1. show spawn time (in minutes) near point icons or mark points which are going to spawn in next N minutes with different color
2. use dialog option to change color/icon for all points which spawned specific selected pokemon in past (maybe with number of spawned pokemons of this type)

**Comment by ximex** [comment]
@azotov i have done all this analytics in QGIS after a export of the SQLite to improved csv files with more infos in it ;-)
here are my sql queries: https://gist.github.com/ximex/18eb879e8c46e74c40d2688fe7981cf7

**Comment by Sejadis** [comment]
@ximex having no clue about anything sql - are there differences in queries for sqlite /mysql? tried to run those queries on the mysql db but everything but the first on gives an error (Error Code: 1248. Every derived table must have its own alias  and  Error Code: 1064. You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '(disappear_time, '-15 minute'), 15, 5) || '-XX:' || SUBSTR(disappear_time, 15, 5' at line 3)

**Comment by ximex** [comment]
@Sejadis yes SQLite and mysql have a different syntax. If i switch to mysql as DB backend (will happen in the next days) i fix the sql queries for mysql/mariadb

**Comment by commanderjsoc** [comment]
This would be amazing if this was added @JZ6

**Comment by commanderjsoc** [comment]
At this time where we need to change how our searchers do their work, I want to recommend this as the most efficient and fasted way to do the most work in the least amount of time possible
